lib/active_model/serializer/adapter/json.rb

```diff
@@ -47,6 +31,32 @@ def serializable_hash(options = {})
       def fragment_cache(cached_hash, non_cached_hash)
         Json::FragmentCache.new().fragment_cache(cached_hash, non_cached_hash)
       end
+
+      private
+
+      def populate_hash_for_array_serializer(name, association, opts)
+        return unless association.respond_to?(:each)
```

Comment:
The nested conditional itself still exists, and now it will run twice the same conditional to each interaction. 
The duplicated line itself is a bad smell. [link1](http://www.laputan.org/pub/patterns/fowler/smells.pdf) [link2](https://www.cqse.eu/publications/2010-code-similarities-beyond-copy-paste.pdf)

> Such cloned code is considered harmful for
two reasons: (1) multiple, possibly unnecessary, duplicates
of code increase maintenance costs [16, 24] and, (2) inconsistent
changes to cloned code can create faults and, hence,
lead to incorrect program behavior [13].

I totally understand the motivation here :smile: but I disagree that keeping the conditional inside ```serializable_hash``` defeats the purpose of this PR, creating the private functions indeed help to keep it clean and organised.